### ISSUE TITLE
Submission is successful without a project

### ISSUE THREAD

**Issue by mathstuf** [Kitware/CDash#502]
I'm seeing this in a CTest script output:

```
Submit files (using https)
   Send to track: master-superbuild
   Using HTTP submit method
   Drop site:https://open.cdash.org
   Uploaded: /Users/dashboard/Dashboards/buildbot/smtk-junction-osx-shared-release_newparaview_superbuild/build/Testing/20170323-0802/Update.xml
   Submission successful
```

Note that there is no `project=…` in the drop site. The reason it isn't set is that the target project isn't known until after the configure step.

Full output is [here](https://buildbot.kitware.com/builders/smtk-junction-osx-shared-release%2Bnewparaview%2Bsuperbuild/builds/310/steps/build-n-test/logs/stdio).

Cc: @bradking @zackgalbreath

**Comment by zackgalbreath** [comment]
[submit.php checks that an existing project was specified](https://github.com/Kitware/CDash/blob/master/public/submit.php#L56-L67).  But in this case, you're not hitting submit.php at all.

Compare the chunk of the log you posted above:
```
   Drop site:https://open.cdash.org
```

To a successful submission later in the same log:

```
   Drop site:https://open.cdash.org/submit.php?project=SMTK
```

In the former case, it's hitting open.cdash.org with no further path defined.  This redirects to index.php, which then redirects to viewProjects.php because no project was specified.

I guess we have to figure out why CTest is removing `submit.php` from the DropSite.

**Comment by mathstuf** [comment]
`CTEST_DROP_LOCATION` is not available until after the configure happens because the submission project differs based on what settings are enabled, so it isn't being removed so much as not being set in the first place.

Why does the home page accept `POST` requests from clients?

**Comment by zackgalbreath** [comment]
> Why does the home page accept POST requests from clients?

No good reason that I can think of.  I guess it's just never been a problem before.

**Comment by williamjallen** [comment]
This issue appears to be resolved as of CDash 3.2.
